sky/backends/backend_utils.py

```diff
@@ -383,6 +383,7 @@ def _get_generated_config(cls, autogen_comment: str, host_name: str,
               Port 22
               {proxy}
             """.rstrip())
+        codegen = codegen + '\n' if not codegen.endswith('\n') else codegen
```

Comment:
Why do we need this if? When will each case happen? If we do the above do we still need it?